include/bls12-381/scalar.hpp

```diff
@@ -215,21 +215,34 @@ void rsh(std::array<uint64_t, N>& out, const std::array<uint64_t, N>& in, uint64
 
 void bn_divn_low(uint64_t *c, uint64_t *d, uint64_t *a, int sa, uint64_t *b, int sb);
 
+template<size_t N, size_t M>
+void bn_divn_safe(std::array<uint64_t, N>& c, std::array<uint64_t, M>& d, const std::array<uint64_t, N>& a, const std::array<uint64_t, M>& b) {
+    
+    static_assert(N >= M, "dividend must be at least same word size as modulus");
+
+    std::array<uint64_t, N+2> modulus = {0};
```

Comment:
You don't need to specify the `0` to get value initialization to all zeroes (see [this case (2)](https://en.cppreference.com/w/cpp/language/zero_initialization)

```suggestion
    std::array<uint64_t, N+2> modulus = {};
```